qiskit/pulse/commands/pulse_decorators.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+# pylint: disable=redefined-builtin
+
+"""
+Pulse decorators.
+"""
+
+import numpy as np
+
+from qiskit.pulse.commands import SamplePulse
+from qiskit.pulse import CommandsError
+
+
+def function(func):
+    """A decorator for generating SamplePulse from python callable.
+
+    Args:
+        func (callable): A function describing pulse envelope.
+    Raises:
+        CommandsError: when invalid function is specified.
+    """
+    def to_pulse(name, duration, *args, **kwargs):
```

Comment:
I think we should make name optional and explicitly evident in signature as reserved kwarg with signature
```python
def to_pulse(duration, *args, name=None, **kwargs):
    ...
```
This way users are not forced to supply a name.